src/main/kotlin/model/Interventions.kt

```diff
@@ -0,0 +1,146 @@
+        Tags.WEB_BROWSER.addTo(this)
+        CloudPlatform.kubernetes.add(this)
+      }
+
+      database = system.addContainer(
+        "Intervention database",
+        "Authoritative source for dynamic framework interventions, service categories, complexity levels; " +
+          "Potential source for dynamic framework providers",
+        "likely PostgreSQL"
+      ).apply {
+        service.uses(this, "connects to", "JDBC")
+        Tags.DATABASE.addTo(this)
+        CloudPlatform.rds.add(this)
+      }
+
+      queue = system.addContainer(
+        "Intervention queue(s)",
+        "Queue(s) for notifications about intervention domain events, please see link for details",
+        "Amazon Simple Queue Service (SQS)"
+      ).apply {
+        setUrl("https://dsdmoj.atlassian.net/wiki/spaces/IC/pages/2461827117/Architecture+overview+-+Interventions")
+        service.uses(this, "publishes domain events to", "SNS")
+        Tags.QUEUE.addTo(this)
+        CloudPlatform.sqs.add(this)
+      }
+
+      translator = system.addContainer(
+        "Intervention-probation translator",
+        "Maintains contacts, appointments and registrations based on dynamic framework intervention domain events",
+        "undefined"
+      ).apply {
+        uses(queue, "consumes dynamic framework intervention domain events from")
+        CloudPlatform.kubernetes.add(this)
+      }
+
+      collector = system.addContainer(
+        "Intervention data collector",
+        "Collects domain events and intervention data for hand-off to the Analytical Platform",
+        "undefined"
+      ).apply {
+        uses(queue, "consumes dynamic framework intervention domain events from")
+        uses(database, "reads snapshots of the intervention data from")
+        CloudPlatform.kubernetes.add(this)
+      }
+
+      system.containers.forEach { Tags.PLANNED.addTo(it) }
+    }
+
+    override fun defineRelationships() {
+      defineAuthentication()
+      defineSharing()
+      defineUsers()
+    }
+
+    fun defineAuthentication() {
+      service.uses(HMPPSAuth.app, "authenticates, authorises users and requests access tokens from", "OAuth2/JWT")
+    }
+
+    fun defineSharing() {
+      service.uses(Delius.communityApi, "retrieves list of dynamic framework providers and probation regions from", "REST/HTTP")
+      service.uses(Delius.communityApi, "retrieves current service user appointments and sentence details from", "REST/HTTP")
+      service.uses(Delius.offenderSearch, "searches service user by identity and gets basic identification details from", "REST/HTTP")
+      service.uses(OASys.assessmentsApi, "retrieves service user current risks and needs from", "REST/HTTP")
+
+      translator.uses(Delius.communityApi, "maintains contacts, appointments, registrations with", "REST/HTTP")
+      collector.uses(AnalyticalPlatform.landingBucket, "pushes intervention data daily to")
+    }
+
+    fun defineUsers() {
```

Comment:
The users also interact directly with HMPPS-Auth to authenticate and authorise. Not sure if we want to represent this?

Reply:
I'm happy either way. What is your preference?

Reply:
Showing the direct interaction is more accurate so if it doesn't confuse things too much I'd prefer we did that

Reply:
done in 10ddd36